File: scala-source/com/barrybecker4/simulation/predprey/creatures/Foxes.scala
// Copyright by Barry G. Becker, 2016-2017. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.simulation.predprey.creatures

import java.awt._


/**
  * Everything we need to know about the current fox population.
  * Foxes are the predators in the simulation.
  * @author Barry Becker
  */
object Foxes {
  val COLOR: Color = Color.RED
  val INITIAL_NUM_FOXES = 100
  private val FOX_BIRTH_RATE = 1.2
  private val FOX_DEATH_RATE = 10
}

class Foxes extends Population {
  override def getName = "Fox"
  override def getInitialPopulation: Int = Foxes.INITIAL_NUM_FOXES
  override def getInitialBirthRate: Double = Foxes.FOX_BIRTH_RATE
  override def getInitialDeathRate: Double = Foxes.FOX_DEATH_RATE
}

File: source/com/barrybecker4/simulation/predprey1/creatures/Foxes.java
/** Copyright by Barry G. Becker, 2000-2011. Licensed under MIT License: http://www.opensource.org/licenses/MIT  */
package com.barrybecker4.simulation.predprey1.creatures;

import java.awt.*;

/**
 * Everything we need to know about the current fox population.
 * Foxes are the predators in the simulation.
 *
 * @author Barry Becker
 */
public class Foxes extends Population {

    public static final Color COLOR = Color.RED;

    public static final int INITIAL_NUM_FOXES = 100;
    private static final double FOX_BIRTH_RATE = 1.2;
    private static final double FOX_DEATH_RATE = 10;

    @Override
    public String getName() {
        return "Fox";
    }

    @Override
    public int getInitialPopulation() {
        return INITIAL_NUM_FOXES;
    }
    @Override
    public double getInitialBirthRate() {
        return FOX_BIRTH_RATE;
    }
    @Override
    public double getInitialDeathRate() {
        return FOX_DEATH_RATE;
    }

}
